analyzer/tests/unit/test_checker_handling.py

```diff
@@ -0,0 +1,204 @@
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+"""
+Test the handling of implicitly and explicitly handled checkers in analyzers
+"""
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+import unittest
+
+from codechecker_analyzer.analyzers.clangsa.analyzer import ClangSA
+from codechecker_analyzer.analyzers.clangtidy.analyzer import ClangTidy
+
+from codechecker_analyzer.buildlog import log_parser
+
+
+class MockContextSA(object):
+    path_env_extra = None
+    ld_lib_path_extra = None
+    checker_plugin = None
+    analyzer_binaries = {'clangsa': 'clang'}
+    compiler_resource_dir = None
+    checker_config = {'clangsa_checkers': ['a.b']}
+    available_profiles = ['profile1']
+    package_root = './'
+
+
+def create_analyzer_sa():
+    # - Begin create config handler
+    # -- Begin create args
+
+    args = []
+
+    # -- End create args
+    # -- Begin create context
+
+    context = MockContextSA()
+
+    # -- End create context
+
+    cfg_handler = ClangSA.construct_config_handler(args, context)
+
+    # - End create config handler
+    # - Begin create build action
+
+    action = {
+        'file': 'main.cpp',
+        'command': "g++ -o main main.cpp",
+        'directory': '/'}
+    build_action = log_parser.parse_options(action)
+
+    # - End create build action
+
+    return ClangSA(cfg_handler, build_action)
+
+
+def create_result_handler(analyzer):
```

Comment:
This function can be simplified like this:
```py
def create_result_handler(analyzer):
    """ Create result handler for construct_analyzer_cmd call. """
    rh = analyzer.construct_result_handler(
        analyzer.buildaction,
        build_action.directory,
        None,
        None)
    rh.analyzed_source_file = build_action.source
    return rh
```

Reply:
Thanks! It is more clear this way. ❇️